docs/migrating-from-product-to-midstream.md

```diff
@@ -0,0 +1,65 @@
+Migration from OpenShift Serverless operator installed using product catalog to the same version using the midstream catalog:
+
+Notes: 
+- This document is Knative Eventing and Kafka specific.
+- Instructions here are for 1.24 of the operator. While procedure is same for other versions, the version numbers and job names will be different.
+
+```shell
+
+# 1. delete Product subscription
+oc delete subscriptions.operators.coreos.com -n openshift-serverless serverless-operator
+
+# 2. delete ClusterServiceVersion
+# This step deletes the pods in `openshift-serverless`, these aren't critical since they provide a way to install Knative
+# components, that means that until last step is complete, installation and configuration functionality on KnativeServing,
+# KnativeEventing, and KnativeKafka resources won't work but Knative components will be up and running as previously.
+oc delete csv -n openshift-serverless serverless-operator.v1.24.0
+
+# 3. delete some jobs that the new operator installation cannot modify (immutable image) - don't worry they will be recreated
+oc delete job -n knative-eventing kafka-controller-post-install-1.24.0              --ignore-not-found
+oc delete job -n knative-eventing knative-kafka-storage-version-migrator-1.24.0     --ignore-not-found
+oc delete job -n knative-eventing storage-version-migration-eventing-eventing-1.3.2 --ignore-not-found
```

Comment:
Can we use label selectors for deleting these jobs? Because otherwise, these commands need to be tweaked based on the "current" version.

Reply:
Done. Please have a look again.

I think we can just do `k delete job -n knative-eventing --all` but let's initially focus on 1.24.